### PR TITLE
Test/e2e

### PR DESCRIPTION
**Pull request by velimd**
included functional test for media viewer

### CODE REVIEW

File: e2e/protractor.conf.js [hunk 1]
```diff
@@ -2,14 +2,23 @@
 // https://github.com/angular/protractor/blob/master/lib/config.ts
 
 const { SpecReporter } = require('jasmine-spec-reporter');
+var fs = require('fs');
```
Comment (linusnorton) [inline e2e/protractor.conf.js]: any reason you went for a var over const?

File: e2e/src/scenarios/download.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+
+  it('should download the pdf', () => {
+    const file = 'e2e/src/downloads/example.pdf';
+
+    page.getPdfViewer().click();
+
+    const downloadPage = page.getDownloadButton();
+    downloadPage.click()
+      .then(() => page.wait(3000))
```
Comment (linusnorton) [inline e2e/src/scenarios/download.e2e-spec.ts]: should we use await rather than .then?

Comment (velimd) [reply e2e/src/scenarios/download.e2e-spec.ts]: youre right, then again, i dont think the then should be there as page.wait would do the same thing

File: e2e/protractor.conf.js (hunk 1 above)
Comment (velimd) [reply e2e/protractor.conf.js]: nope, just was looking into how to read files in typescript. will change it.

File: e2e/src/scenarios/download.e2e-spec.ts (hunk 2 above)
Comment (linusnorton) [reply e2e/src/scenarios/download.e2e-spec.ts]: Well the .then comment applies to the whole PR really. I don't think we should be using them when we have `await`. 

The page.wait function will probably still be needed as that's to wait for a specific amount of time (I assume to let the HTML update).

Comment (velimd) [reply e2e/src/scenarios/download.e2e-spec.ts]: yh for html update and in this case its to wait for the file to be downloaded

File: e2e/protractor.conf.js [hunk 3]
```diff
@@ -24,5 +33,23 @@ exports.config = {
...
     });
     jasmine.getEnv().addReporter(new SpecReporter({ spec: { displayStacktrace: true } }));
+    removePreviousDownloads();
   }
 };
+
+function removePreviousDownloads() {
+  try {
+    const downloadsPath = 'e2e/src/downloads';
+    const files = fs.readdirSync(downloadsPath);
```
Comment (JeroenRijks) [inline e2e/protractor.conf.js]: Is this an array of relative file paths, or an array of actual File objects? If it's the first, could you rename to something like relativeFilePaths?

File: e2e/src/app.po.ts [hunk 4]
```diff
@@ -1,27 +1,124 @@
...
+
+  getCurrentZoomOption() {
+    return element(by.id('customScaleOption'));
+  }
+
+  getZoomSelect() {
+    return element(by.id('scaleSelect'));
+  }
+
+  selectZoomValue(zoomOption) {
```
Comment (JeroenRijks) [inline e2e/src/app.po.ts]: this, and other methods here, are still getters right? getSelectZoomValue?

File: e2e/src/scenarios/print.e2e-spec.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    const printPage = page.getPrintButton();
+    printPage.click();
+
+    // clean up by moving to app.po
+    browser.getAllWindowHandles().then(function(handles) {
+      expect(handles.length).toEqual(2);
+      browser.switchTo().window(handles[1]);
+      expect(page.getPrintDialog).toBeTruthy();
+      page.wait(1000);
+      browser.close();
```
Comment (JeroenRijks) [inline e2e/src/scenarios/print.e2e-spec.ts]: weird how you close the browser, then switch to another tab in the browser. Is there any way to do something like `browser.close(window(handles[1]))`?

Comment (JeroenRijks) [reply e2e/src/scenarios/print.e2e-spec.ts]: nvm, I couldn't find anything better than what you've got

File: e2e/protractor.conf.js (hunk 3 above)
Comment (velimd) [reply e2e/protractor.conf.js]: it returns an array with the files names rather than paths.

**Pull request merged**